File: hw/caizapanta/u3/HW17Singleton/src/ec/edu/espe/model/USTax.java
package ec.edu.espe.model;

/**
 * 
 * @author Tammy Caizapanta
 */

public class USTax {
    private static USTax instance;
    private float taxValue;

    private USTax() {
        taxValue = 0.08f;
    }

    public static USTax getInstance() {
        if (instance == null) {
            instance = new USTax();
        }
        return instance;
    }

    public float salesTotal(float basePrice) {
        return basePrice + (basePrice * taxValue);
    }

    public float getTaxValue() {
        return taxValue;
    }

    public void setTaxValue(float taxValue) {
        this.taxValue = taxValue;
    }
}

File: hw/paillacho/U3/HW18Singleton/CalculatorPython/model/USTAX.py
# ec/edu/espe/model/us_tax.py
# Author: Carlos Paillacho
class USTax:
    __instance = None

    def __init__(self):
        if USTax.__instance is not None:
            raise Exception("This class is a singleton! Use get_instance().")
        self.tax_value = 0.08
        USTax.__instance = self

    @staticmethod
    def get_instance():

        if USTax.__instance is None:
            USTax()
        return USTax.__instance

    def sales_total(self, base_price: float) -> float:

        return base_price + (base_price * self.tax_value)

    def get_tax_value(self) -> float:
        return self.tax_value

    def set_tax_value(self, tax_value: float):
        self.tax_value = tax_value
